Sketch module organisation

jorisdral opened the pull request.

dcoutts commented on src/Database/LSMTree/Run/Index/Compact.hs:
```diff
@@ -0,0 +1 @@
+module Database.LSMTree.Run.Index.Compact () where
```
We only plan to have one index, so I don't think we need extra hierarchy here.
```suggestion
module Database.LSMTree.Run.IndexCompact () where
```
or can just call it `Database.LSMTree.Run.Index`

jorisdral replied:
In the future, might it not make sense to have a full index variant too? Not for the Cardano use case, but for the more general use case

dcoutts replied:
True, but even there we probably don't need a lot of internal hierarchy.

dcoutts reviewed (commented):
To help us think about what goes where, lets put a sentence or so as the module header documentation, e.g.
```
-- | Implementation of the K-way file merge.
--
module Database.LSMTree.KMerge () where
```

Though perhaps we should also relate this to the WP items somewhat, though not as haddock docs, perhaps just TODOs. For example for the above, does this include the pure bits and the I/O or just one of them?

jorisdral commented:
Thanks for the review. I've made some changes that can now be reviewed again. Most notably, I moved the internal modules into an `Internal` module hierarchy, because I don't mind the indirection and it means that implementation modules and public API modules are clearly separated, which I prefer. Let me know if you disagree

> Though perhaps we should also relate this to the WP items somewhat, though not as haddock docs, perhaps just TODOs. For example for the above, does this include the pure bits and the I/O or just one of them?

I've added  module-header-level comments inside TODO sections

dcoutts reviewed (approved):
LGTM
